crypto/bio/bss_dgram_pair.c

```diff
@@ -988,8 +1130,14 @@ static size_t dgram_pair_write_inner(struct bio_dgram_pair_st *b, const uint8_t
          * ringbuf and read from the peer ringbuf.
          */
         ring_buf_head(&b->rbuf, &dst_buf, &dst_len);
-        if (dst_len == 0)
-            break;
+        if (dst_len == 0) {
+            if (b->peer != NULL) /* dgram_pair is not resizeable dgram_mem is */
+                break;
+            /* just double the size */
+            if (!ring_buf_resize(&b->rbuf, b->req_buf_len * 2))
```

Comment:
There is research indicating that doubling is too much and that something close to but below the golden ratio is better.  We use 8/5 as the scaling factor in several other places.  E.g.

https://github.com/openssl/openssl/blob/30667f5c306dbc11ac0e6fddc7d26fd984d546ab/crypto/stack/stack.c#L140-L174

We should be consistent and use the same scaling here too.